From fc207f85521c3939600bd3e5a337c6ed2dfaa950 Mon Sep 17 00:00:00 2001
From: Michael Natterer <mitch@gimp.org>
Date: Mon, 8 Feb 2010 23:47:08 +0100
Subject: [PATCH] process: don't leak each node set on a GeglProcessor

At some point a g_value_get_boxed() was bogusly replaced by a
g_value_dup_boxed(). Also added notifications for "node" and
"rectangle".
---
 gegl/process/gegl-processor.c |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/gegl/process/gegl-processor.c b/gegl/process/gegl-processor.c
index be90881..8d5e2fb 100644
--- a/gegl/process/gegl-processor.c
+++ b/gegl/process/gegl-processor.c
@@ -194,7 +194,7 @@ gegl_processor_set_property (GObject      *gobject,
   switch (property_id)
     {
       case PROP_NODE:
-        gegl_processor_set_node (self, g_value_dup_object (value));
+        gegl_processor_set_node (self, g_value_get_object (value));
         break;
 
       case PROP_CHUNK_SIZE:
@@ -278,6 +278,8 @@ gegl_processor_set_node (GeglProcessor *processor,
     }
 
   g_object_ref (processor->input);
+
+  g_object_notify (G_OBJECT (processor), "node");
 }
 
 
@@ -354,6 +356,8 @@ gegl_processor_set_rectangle (GeglProcessor       *processor,
       gegl_operation_context_set_need_rect   (processor->context,
                                               &processor->rectangle);
     }
+
+  g_object_notify (G_OBJECT (processor), "rectangle");
 }
 
 /* Will generate band_sizes that are adapted to the size of the tiles */
-- 
1.6.6

